TACC-Cloud/hazmapper
# task/WG-208 Listing Dropdown

## Overview: ##

## PR Status: ##


## Related Jira tickets: ##

* [WG-208](https://tacc-main.atlassian.net/browse/WG-208)

## Summary of Changes: ##

- Added a SystemSelect dropdown that shows the user systems available to them 
- Added placeholder for designsafe projects which will be added when useProject hook is avaiable
- Added a useSystems hook that retrieves designsafe tapis systems 
- Added tests for SystemSelect with sample fixture 
- Add types for tapis System response
- Added SystemSelect component and event handling to MainMenu for demo purposes

## Testing Steps: ##
1. Get the react project up and running and go to localhost:4200 to see the SystemSelect dropdown

## UI Photos:

<img width="471" alt="Screen Shot 2024-01-19 at 1 44 54 PM" src="https://github.com/TACC-Cloud/hazmapper/assets/23081932/3291108c-fef3-4e38-b3b1-46a8f4632f5b">

<img width="471" alt="Screen Shot 2024-01-19 at 1 45 01 PM" src="https://github.com/TACC-Cloud/hazmapper/assets/23081932/7c7bf413-7174-4c8e-af69-977b76111acb">


## Notes: ##

TODO: 

* [ ] Add Designsafe projects to SystemSelect as well using project hook
